src/msg.hpp

```diff
@@ -39,6 +39,9 @@
 #include "atomic_counter.hpp"
 #include "metadata.hpp"
 
+//  bits 2-5
+#define CMD_TYPE_MASK 28
```

Comment:
maybe define this as a hex literal instead to make the relationship to the bits more obvious?

In c++ style, a constant would be preferable to a preprocessor definition.